File: src/insertion_sort.rs
fn shift_array(array: &mut [usize], start: usize, end: usize) {
    for i in (start..end).rev() {
        array.swap(i, i + 1);
    }
}

/// Sorts the given array, in place, by using the Insertion sort algorithm.
pub fn sort_by_insertion_sort(array: &mut [usize]) {
    unimplemented!();

    for index in 1..array.len() {
        println!("{:?}", array);

        let current_element = array[index];

        let mut i = index - 1;

        if array[i] < current_element {
            continue;
        }

        while i != 0 && array[i] > current_element {
            i -= 1;
        }

        println!("{}", i);

        shift_array(array, i, index);
    }
}

File: python_src/insertion_sort.py
def shift_array(array, start, end):
    for i in range(end, start, -1):
        array[i], array[i - 1] = array[i - 1], array[i]

def sort_by_insertion_sort(array):
    for i in range(1, len(array)):
        curr_element = array[i]

        new_index = i - 1

        while new_index != -1 and array[new_index] > curr_element:
            new_index -= 1

        new_index += 1

        shift_array(array, new_index, i)